source/server/async_operation.h

```diff
@@ -0,0 +1,74 @@
+#ifndef NIDEVICE_GRPC_DEVICE_ASYNC_OPERATION_H
+#define NIDEVICE_GRPC_DEVICE_ASYNC_OPERATION_H
+
+#include <grpcpp/alarm.h>
+#include <grpcpp/grpcpp.h>
+
+#include <deque>
+#include <memory>
+#include <mutex>
+#include <queue>
+
+namespace nidevice_grpc {
+
+template <typename TResponse, typename TJoinable>
```

Comment:
TODO: I'm using "thread"/TJoinable in this api. But the in-product implementation uses a "callback registration". I was thinking that I might use a `join` method to ensure that it was detached, but that's just happening in the destructor (hence `AutoJoinThread`).

I think `producer` or `registration` would be better terms.